csi/exporter.py

```diff
@@ -67,12 +83,33 @@ def collect(self):
                 inodes_free_count.add_metric(labels, free)
                 inodes_used_count.add_metric(labels, used)
 
+                # Gathers capacity metrics for each subvol
+                pvcpath = pathlib.Path(os.path.join(pth, PV_TYPE_SUBVOL)).glob("*/*/*")
+                for subvol in pvcpath:
+                    if pathlib.Path(subvol).is_dir():
+                        pvcname = os.path.basename(subvol)
+                        pvclabels = labels + [pvcname]
+                        try:
+                            # Get extended attributes for pvc
+                            size = os.getxattr(subvol, "trusted.gfs.squota.size")
+                            limit = os.getxattr(subvol, "trusted.gfs.squota.limit")
+                            subvol_capacity_bytes.add_metric(pvclabels, limit)
+                            subvol_capacity_used_bytes.add_metric(pvclabels, size)
+                            # Convert bytes type into int to compute the free capacity of the subvol
+                            free = int(limit.decode("utf-8")) - int(size.decode("utf-8"))
+                            subvol_capacity_free_bytes.add_metric(pvclabels, free)
+                        except Exception:
+                            pass
```

Comment:
- Only `pass` an exception in the worst case
- Can you please log an error here?

Reply:
I deleted the code block that retrieved the xattr from the pvc, so it's not useful anymore.